auto_round/autoround.py

```diff
@@ -664,94 +666,66 @@ def get_block_forward_func(self, name):
             function: The forward function.
         """
 
-        def forward(m, hidden_states, *positional_args, **kwargs):
+        def forward(m, hidden_states=None, *positional_inputs, **kwargs):
             """Rewrite forward function, process and collect input data.
 
             Args:
                 hidden_states (torch.Tensor): The hidden states tensor.
-                *positional_args: Variable number of positional arguments.
+                *positional_inputs: Variable number of positional arguments.
                 **kwargs: Variable number of keyword arguments.
 
             Returns:
                 NotImplementedError: Getting the first layer inputs and then raise the error to save runtime.
             """
-            if self.share_attention_mask_flag is None:
-                self.input_dim = check_hidden_state_dim(self.model, positional_args)
-                self.share_attention_mask_flag = check_share_attention_mask(self.model, hidden_states, **kwargs)
-                self.not_share_position_ids_flag = check_not_share_position_ids(self.model, **kwargs)
-                self.not_share_rotary_pos_emb_flag = check_not_share_rotary_pos_emb(self.model, **kwargs)
-            if name in self.inputs:
-                if self.train_bs == 1 and self.not_share_rotary_pos_emb_flag:
-                    self.inputs[name]["input_ids"].append(hidden_states.to("cpu"))
-                else:
-                    self.inputs[name]["input_ids"].extend(
-                        list(torch.split(hidden_states.to("cpu"), 1, dim=self.input_dim)))
-            else:
+            if name not in self.inputs:
                 self.inputs[name] = {}
-                if self.train_bs == 1 and self.not_share_rotary_pos_emb_flag:
-                    self.inputs[name]["input_ids"] = [hidden_states.to("cpu")]
-                else:
-                    self.inputs[name]["input_ids"] = list(torch.split(hidden_states.to("cpu"), 1, dim=self.input_dim))
-
-            if "positional_inputs" not in self.inputs[name]:
-                self.inputs[name]["positional_inputs"] = []
-            for idx, item in enumerate(positional_args):
-                self.inputs[name]["positional_inputs"] = to_device(positional_args)
+                self.batch_size = check_model_batch(self.model, self.batch_size)
```

Comment:
if batch_size has been changed, gradient_accumulate_steps should be changed correspondingly. Also dump a warning  to inform user these values  has been changed